toolchain/lc-compile/src/emit.cpp

```diff
@@ -1719,3 +1729,253 @@ void OutputEnd(void)
     
     fclose(s_output);
 }
+
+//////////
+
+struct DependMapping
+{
+    NameRef module;
+    const char *source;
+    char *interface;
+    time_t source_time;
+    time_t interface_time;
+    bool is_interface;
+    bool processed;
+    bool changed;
+};
+
+struct DependDependency
+{
+    NameRef module;
+    NameRef dependency;
+};
+
+int DependencyMode = 0;
```

Comment:
I think perhaps DependencyMode should be an enum in order to avoid magic numbers.